packages/api/tests/endpoints/test-executions.js

```diff
@@ -2,78 +2,266 @@
+  // we need this because a granule has a foreign key referring to collections
+  const collectionName = 'fakeCollection';
+  const collectionVersion = 'v1';
+
+  t.context.testCollection = fakeCollectionFactory({
+    name: collectionName,
+    version: collectionVersion,
+    duplicateHandling: 'error',
+  });
+  const dynamoCollection = await collectionModel.create(
+    t.context.testCollection
+  );
+  t.context.collectionId = constructCollectionId(
+    dynamoCollection.name,
+    dynamoCollection.version
+  );
+
+  const testPgCollection = fakeCollectionRecordFactory({
+    name: collectionName,
+    version: collectionVersion,
+  });
+
+  [t.context.collectionCumulusId] = await collectionPgModel.create(
+    knex,
+    testPgCollection
+  );
+
+  await esClient.indices.refresh();
+});
+
+test.beforeEach(async (t) => {
+  const { esAlias, knex } = t.context;
+
+  const workflowName1 = cryptoRandomString({ length: 6 });
+  const workflowName2 = cryptoRandomString({ length: 6 });
+
+  // create fake Postgres executon records
+  t.context.fakePGExecutions = [
+    fakeExecutionRecordFactory({ workflow_name: workflowName1 }),
+    fakeExecutionRecordFactory({ workflow_name: workflowName2 }),
+  ];
+
+  [t.context.executionCumulusId1, t.context.executionCumulusId2]
+    = await Promise.all(
+      t.context.fakePGExecutions.map((execution) =>
+        executionPgModel.create(knex, execution))
+    );
+
+  const granuleId1 = randomId('granuleId1');
+  const granuleId2 = randomId('granuleId2');
+
+  // create fake Dynamo granule records
+  t.context.fakeGranules = [
+    fakeGranuleFactoryV2({ granuleId: granuleId1, status: 'completed' }),
+    fakeGranuleFactoryV2({ granuleId: granuleId2, status: 'failed' }),
+  ];
+
+  await Promise.all(
+    t.context.fakeGranules.map((granule) =>
+      granuleModel
+        .create(granule)
+        .then((record) => indexer.indexGranule(esClient, record, esAlias)))
+  );
+
+  // create fake Postgres granule records
+  t.context.fakePGGranules = [
+    fakeGranuleRecordFactory({
+      granule_id: granuleId1,
+      status: 'completed',
+      collection_cumulus_id: t.context.collectionCumulusId,
+    }),
+    fakeGranuleRecordFactory({
+      granule_id: granuleId2,
+      status: 'failed',
+      collection_cumulus_id: t.context.collectionCumulusId,
+    }),
+  ];
+
+  [t.context.granuleCumulusId] = await Promise.all(
+    t.context.fakePGGranules.map((granule) =>
+      granulePgModel.create(knex, granule))
+  );
+
+  t.context.joinRecords = [
+    {
+      execution_cumulus_id: t.context.executionCumulusId1[0],
+      granule_cumulus_id: Number(t.context.granuleCumulusId),
+    },
+    {
+      execution_cumulus_id: t.context.executionCumulusId2[0],
+      granule_cumulus_id: Number(t.context.granuleCumulusId),
+    },
+  ];
+
+  await Promise.all(
+    t.context.joinRecords.map((joinRecord) =>
+      granulesExecutionsPgModel.create(knex, joinRecord))
+  );
```

Comment:
Use the `upsertGranuleWithExecutionJoinRecord` helper instead of having to create the records and join records separately

Reply:
Ohh thanks for the tip!